client/iface/wgproxy/udp/proxy.go

```diff
@@ -193,3 +192,15 @@ func (p *WGUDPProxy) proxyToLocal(ctx context.Context) {
 		}
 	}
 }
+
+func (p *WGUDPProxy) remoteConnRead(ctx context.Context, buf []byte) (n int, err error) {
+	n, err = p.remoteConn.Read(buf)
+	if err != nil {
+		if ctx.Err() != nil {
+			return
```

Comment:
This should rather return `n, nil` and the logging in line 202 should be done in the upper layer.

Reply:
I made this function because of the Sonarqube restriction. If I handle the error in upper layer then come code complexity issue.